File: 31_naive_pattern_search.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 31: Implement the Naive pattern searching algorithm.
Reads two lines: text and pattern; prints all starting indices (0-based) separated by spaces.
*/

int main()
{
    string t, p;
    if (!getline(cin, t))
        return 0;
    if (!getline(cin, p))
        return 0;
    vector<int> res;
    int n = t.size(), m = p.size();
    for (int i = 0; i + m <= n; ++i)
    {
        int j = 0;
        while (j < m && t[i + j] == p[j]) ++j;
        if (j == m)
            res.push_back(i);
    }
    for (size_t i = 0; i < res.size(); ++i)
    {
        if (i)
            cout << ' ';
        cout << res[i];
    }
    cout << '\n';
    return 0;
}

// Time Complexity: O(n*m)
// Space Complexity: O(1)

File: 31_naive_pattern_search.py
"""
Task 31: Implement the Naive pattern searching algorithm.
Reads two lines: text and pattern; prints all starting indices (0-based) separated by spaces.
"""
import sys

if __name__ == '__main__':
    t = sys.stdin.readline().rstrip('\n')
    p = sys.stdin.readline().rstrip('\n')
    res = []
    n, m = len(t), len(p)
    for i in range(n-m+1):
        if t[i:i+m] == p: res.append(str(i))
    print(' '.join(res))

# Time Complexity: O(n*m)
# Space Complexity: O(1)
